### PR TITLE
Improve import formatting

### PR DESCRIPTION
**Pull request by rbradford**
- **misc: Adapt consistent import style formatting**
- **build: Make cargo fmt check apply to all packages**
- **build: Apply cargo fmt check to fuzz workspace**

### CODE REVIEW

**Comment by rbradford** [comment]
Raced with #6767 but this includes "fuzz" workspace fixes.

**Comment by TimePrinciple** [comment]
> Raced with #6767 but this includes "fuzz" workspace fixes.

Noted, closing #6767

**Comment by TimePrinciple** [comment]
Should we also remove `typo` check in CI since I never saw it gets executed 🤔

**Comment by rbradford** [comment]
> Should we also remove `typo` check in CI since I never saw it gets executed 🤔

I think it is run - https://github.com/cloud-hypervisor/cloud-hypervisor/actions/runs/11082739828/job/30796159673?pr=6768 ? Or is there something more subtle here?

**Review by jinankjain — APPROVED** [review]
Yay! 🎉

**Comment by TimePrinciple** [comment]
> > Should we also remove `typo` check in CI since I never saw it gets executed 🤔
> 
> I think it is run - https://github.com/cloud-hypervisor/cloud-hypervisor/actions/runs/11082739828/job/30796159673?pr=6768 ? Or is there something more subtle here?

Never mind, I had a look on that and it's only triggered when `pull_request`

**Comment by lisongqian** [comment]
We got a warning: https://github.com/cloud-hypervisor/cloud-hypervisor/actions/runs/11082739828/job/30796159759?pr=6768, since `group_imports = StdExternalCrate` can only available in nightly.

**Comment by rbradford** [comment]
> We got a warning: https://github.com/cloud-hypervisor/cloud-hypervisor/actions/runs/11082739828/job/30796159759?pr=6768, since `group_imports = StdExternalCrate` can only available in nightly.

Well spotted - (I use nightly so I can easily test the fuzzer.) I will work out a solution to ensure this is enforced.

File: .github/workflows/formatting.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  build:
+    name: Code Formatting
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rust:
+          - nightly
+        target:
+          - x86_64-unknown-linux-gnu
+          - aarch64-unknown-linux-musl
```
Comment (TimePrinciple) [inline .github/workflows/formatting.yaml:17]: Great work! BTW why are we having `gnu` for `x86_64` and `musl` for `riscv64` :)

Comment (rbradford) [reply .github/workflows/formatting.yaml:17]: These are the most mature toolchains for those two and I didn't want to waste resources with a full set.

**Pull request merged**